gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifyContentsPerExecutor.java

```diff
@@ -267,4 +209,119 @@ private Instant getCutoffTimeForRef(String reference, Map<String, Instant> dropp
                 GCUtil.deserializeReference(reference).getName(),
                 gcParams.getDefaultCutOffTimestamp());
   }
+
+  // --- Methods for computing expired content rows ----------
+  /**
+   * convert scala Iterator of reference into scala Iterator of {@link Row}.
+   *
+   * <p>Each reference can produce one or more rows.
+   */
+  private scala.collection.Iterator<Row> getExpiredContentRows(
+      scala.collection.Iterator<String> references,
```

Comment:
Maybe we can convert the iterator to DF and use a Spark API for transformation, instead of using the `scala.collection.Iterator`?

Reply:
this is in executor and spark context is not available in executor. map partitions need the iterator as output